nbgitpuller/plugins/plugin_helper.py

```diff
@@ -0,0 +1,146 @@
+import subprocess
+import os
+import stat
+import logging
+import requests
+from requests_file import FileAdapter
+import shutil
+import re
+
+
+# for large files from Google Drive
+def get_confirm_token(response):
+    for key, value in response.cookies.items():
+        if key.startswith('download_warning'):
+            return value
+    return None
+
+
+# sets up the a local repo that acts like a remote
+def initialize_local_repo(local_repo_path):
+    logging.info(f"Creating local_repo_path: {local_repo_path}")
+    try:
+        os.makedirs(local_repo_path)
+    except OSError:
+        logging.info(f"Directory exists: {local_repo_path}")
+
+    subprocess_helper("git init --bare", local_repo_path)
+
+    # Make our bare repository serveable over dumb HTTP
+    hook_path = os.path.join(local_repo_path, 'hooks', 'post-update')
```

Comment:
Hmm, why do we need to do this? `git clone` can take a path as source, we don't need it to be over HTTP right?

Reply:
yes, ok: https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols. I didn't quite take in how git clone was going to work as I began to move through the changes. I never re-visited this moment as I the process became clearer ... good deal